src/templater.rs

```diff
@@ -569,17 +569,12 @@ impl<'a> CommitOrChangeId<'a> {
 
     /// The length of the id printed will be the maximum of `total_len` and the
     /// length of the shortest unique prefix
-    pub fn shortest(&self, total_len: i64) -> ShortestIdPrefix {
-        let hex = self.hex();
-        let (prefix, rest) = extract_entire_prefix_and_trimmed_tail(
-            &hex,
-            self.repo.shortest_unique_id_prefix_len(self.as_bytes()),
-            max(total_len, 0) as usize,
-        );
-        ShortestIdPrefix {
-            prefix: prefix.to_string(),
-            rest: rest.to_string(),
-        }
+    pub fn shortest(&self, total_len: usize) -> ShortestIdPrefix {
+        let prefix_len = self.repo.shortest_unique_id_prefix_len(self.as_bytes());
+        let mut prefix = self.hex();
```

Comment:
Nit: I would prefer if you called this `hex` or `result`. It only becomes the prefix several lines down.

Reply:
For this ~5-line code, I prefer not renaming `hex` to `prefix` to build `ShortestIdPrefix`.

Reply:
Actually, code reorder made it less clear, so renamed to `hex`.